### DISCUSSION TITLE
Please tag full releases

### DISCUSSION THREAD

**Discussion by SnarkBoojum** [palantir/blueprint#5746] [Q&A] [answered]
I wanted to see if it was possible to package blueprint for Debian, and found out it would be quite difficult: each and every subpackage has a tag for each release, but there is no big tag for the full stack, so it's difficult to know when there's a coherent whole.

Can you also tag full releases? Thanks!

**Comment by adidahiya** [comment]
Our releases are tagged using [lerna](https://github.com/lerna/lerna), which does not support this kind of "full release" tag AFAIK. We'd have to add extra build tooling to make a tag like `v4.0.0-beta.10` _in addition to_ the current style of tags, `@blueprintjs/core@4.0.0-beta.10`... and I'm not inclined to do that for the minimal benefit it would provide. Note that every time the core package is released, we also trigger a release for every other package, so you can probably just look at the core release tags for your use case.

In summary: Blueprint is a collection of packages and we tag releases in a way that is pretty standard for the NPM ecosystem. You don't need to use all of Blueprint, and the core package version is the most "important" version number.

**Comment by SnarkBoojum** [comment]
Uh... "git tag -s 2021.12.01" is all it takes for a full release ; no extra tooling is necessary.

**Comment by adidahiya** [comment, marked as answer]
I still don't think the semantics you're expecting match up with how this collection of NPM packages work. There's no single "coherent whole" at any given time... many different versions of the different packages are supported at the same time, constrained by dependency ranges which are defined in `package.json` files.

Have you tried to package together any other monorepos in this way? I'm sure you'd run into the same problems elsewhere in the NPM ecosystem.

Even if I _did_ `git tag v4.0.0-beta.10` for the latest core release, it wouldn't be any different from the `@blueprintjs/core@4.0.0-beta.10` tag. So I'd suggest just looking at the core tags.
